src/api/procedures/setVenueFiltering.ts

```diff
@@ -0,0 +1,96 @@
+import { Asset, Procedure } from '~/internal';
+import { SetVenueFilteringParams, TxTags } from '~/types';
+import { BatchTransactionSpec, ProcedureAuthorization } from '~/types/internal';
+import { bigNumberToU64, booleanToBool, stringToTicker } from '~/utils/conversion';
+import { checkTxType } from '~/utils/internal';
+
+/**
+ * @hidden
+ */
+export type Params = {
+  ticker: string;
+} & SetVenueFilteringParams;
+
+/**
+ * @hidden
+ */
+export async function prepareVenueFiltering(
+  this: Procedure<Params, void>,
+  args: Params
+): Promise<BatchTransactionSpec<void, unknown[][]>> {
+  const { context } = this;
+  const {
+    polymeshApi: { tx, query },
+  } = context;
+  const { ticker, enabled, allowedVenues, disallowedVenues } = args;
+  const rawTicker = stringToTicker(ticker, context);
+  const transactions = [];
+
+  const isEnabled = await query.settlement.venueFiltering(ticker);
+
+  if (enabled !== undefined && isEnabled.valueOf() !== enabled) {
+    transactions.push(
+      checkTxType({
+        transaction: tx.settlement.setVenueFiltering,
+        args: [rawTicker, booleanToBool(enabled, context)],
+      })
+    );
+  }
```

Comment:
In a scenario, where I only pass in the allowed venues my expectation is that even if the ticker has not enabled the venue filtering it will automatically enable it. IMO, in case when `enabled` is undefined and `allowedVenues`/`disallowedVenues` are passed, `setVenueFiltering` transaction should also be triggered (given that existing value is `isEnabled` is false

Reply:
To explain a bit more, IMO we should have restrictions on the args such as 
```
{
   enabled: true | undefined, 
   allowedVenues?: BigNumber[], 
   disallowedVenues?: BigNumber[]
} || {
  enabled: false
}
```
With disabling the filtering, I should not be allowed to add/remove venues

Thoughts @polymath-eric @VictorVicente

Reply:
I thought chain would prevent this, but since it allows it maybe we can leave as is